aws-sqs-node/src/worker.ts

```diff
@@ -0,0 +1,111 @@
+// @@@SNIPSTART aws-sqs-node
+import { ReceiveMessageCommand } from '@aws-sdk/client-sqs'
+import { Client, Connection } from '@temporalio/client'
+import { Worker } from '@temporalio/worker'
+import cluster from 'node:cluster'
+import { availableParallelism } from 'node:os'
+import process from 'node:process'
+import * as activities from './activities'
+import { getLogger, getQueueUrl, getSQSClient } from './utils'
+import { helloWorld } from './workflows'
+const sqs = getSQSClient()
+const queueUrl =  getQueueUrl()
+const logger = getLogger()
+
+let temporalClient: Client
+const getTemporalClient = async () => {
+  if (!temporalClient) {
+    const connection = await Connection.connect()
+    temporalClient = new Client({
+      connection,
+    })
+  }
+  return temporalClient
+}
+
+async function pollQueue(): Promise<void> {
+    const receiveMessageCommand = new ReceiveMessageCommand({
+      QueueUrl: queueUrl,
+      AttributeNames: ['All'],
+      VisibilityTimeout: 300,
+      MaxNumberOfMessages: 10,
+    })
+
+    const data = await sqs.send(receiveMessageCommand)
+    if (!data.Messages) {
+      logger.info('No messages in queue')
+      return
+    }
+
+    for (const message of data.Messages) {
+      const { nanoid } = await import('nanoid')
+      // keep in mind, you will have N temporal connections here, 
+      // one for each poller
+      const client = await getTemporalClient()
+      logger.info({ message }, 'Starting workflow')
+      await client.workflow.start(helloWorld, {
+        args: [message],
+        workflowId: nanoid(),
+        taskQueue: 'hello-world-queue',
```

Comment:
```suggestion
        // If a duplicate event might arrive after the Workflow started by the original event has completed, 
        // set this reuse policy to prevent duplicate Workflows:
        // workflowIdReusePolicy: WorkflowIdReusePolicy .WORKFLOW_ID_REUSE_POLICY_REJECT_DUPLICATE,
        taskQueue: 'hello-world-queue',
```